controller/src/command.rs

```diff
@@ -116,6 +117,81 @@ where
 	Ok(())
 }
 
+pub fn rewind_hash<'a, L, C, K>(
+	owner_api: &mut Owner<L, C, K>,
+	keychain_mask: Option<&SecretKey>,
+) -> Result<(), Error>
+where
+	L: WalletLCProvider<'static, C, K>,
+	C: NodeClient + 'static,
+	K: keychain::Keychain + 'static,
+{
+	controller::owner_single_use(None, keychain_mask, Some(owner_api), |api, m| {
+		// Just address at derivation index 0 for now
+		let rewind_hash = api.get_rewind_hash(m)?;
+		println!();
+		println!("Wallet Rewind Hash");
+		println!("-------------------------------------");
+		println!("{}", rewind_hash);
+		println!();
+		Ok(())
+	})?;
+	Ok(())
+}
+
+/// Arguments for View Wallet scan command
+pub struct ViewWalletScanArgs {
+	pub rewind_hash: String,
+	pub start_height: Option<u64>,
+	pub backwards_from_tip: Option<u64>,
+}
+
+pub fn scan_rewind_hash<L, C, K>(
+	owner_api: &mut Owner<L, C, K>,
+	args: ViewWalletScanArgs,
+	dark_scheme: bool,
+) -> Result<(), Error>
+where
+	L: WalletLCProvider<'static, C, K> + 'static,
+	C: NodeClient + 'static,
+	K: keychain::Keychain + 'static,
+{
+	controller::owner_single_use(None, None, Some(owner_api), |api, m| {
+		let rewind_hash = args.rewind_hash;
+		let tip_height = api.node_height(m)?.height;
+		let start_height = match args.backwards_from_tip {
+			Some(b) => tip_height.saturating_sub(b),
+			None => match args.start_height {
+				Some(s) => s,
+				None => 1,
+			},
+		};
+		warn!(
+			"Starting view wallet output scan from height {} ...",
+			start_height
+		);
+		let result = api.scan_rewind_hash(rewind_hash, Some(start_height));
+		let deci_sec = time::Duration::from_millis(100);
+		thread::sleep(deci_sec);
```

Comment:
Just curious as to why that's needed; (not a big deal)

Reply:
It's because of the "updater thread" being too slow.

When running `grin-wallet rewind_hash`
 - With the sleep() :
```
20211212 09:29:34.150 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
20211212 09:29:34.174 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
20211212 09:29:34.175 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning Complete
**20211212 09:29:34.288 WARN grin_wallet_controller::command - View wallet check complete**
```

- WIthout the sleep
```
20211212 09:29:34.150 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
**20211212 09:29:34.288 WARN grin_wallet_controller::command - View wallet check complete**
20211212 09:29:34.174 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
20211212 09:29:34.175 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning Complete
```

Reply:
It's because of the "updater thread" being too slow.

When running `grin-wallet rewind_hash`:

- With the sleep():
```
20211212 09:29:34.150 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
20211212 09:29:34.174 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
20211212 09:29:34.175 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning Complete
**20211212 09:29:34.288 WARN grin_wallet_controller::command - View wallet check complete**
```

- Without:
```
20211212 09:29:34.150 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
**20211212 09:29:34.288 WARN grin_wallet_controller::command - View wallet check complete**
20211212 09:29:34.174 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning - 99% complete
20211212 09:29:34.175 WARN grin_wallet_libwallet::api_impl::owner_updater - Scanning Complete
```